### PR TITLE
Add glob pattern support to -f parameter

### PR DESCRIPTION
**Pull request by hanabi1224**
https://github.com/codecov/codecov-exe/issues/68

### CODE REVIEW

**Comment by hanabi1224** [comment]
Not sure why there's no appveyor ci build for this PR but I have one here:
https://ci.appveyor.com/project/hanabi1224/codecov-exe/builds/24817390/job/vxwhn3edaxqtnc7w

File: Source/Codecov.Tests/Coverage/Tool/CoverageTests.cs [hunk 1]
```diff
@@ -10,19 +10,19 @@ namespace Codecov.Tests.Coverage.Tool
 {
     public class CoverageTests
     {
-        [Theory, InlineData(null), InlineData(""), InlineData("./ coverage.xml")]
+        [Theory, InlineData(null), InlineData(""), InlineData("./coverage.xml")]
```
Comment (AdmiringWorm) [inline Source/Codecov.Tests/Coverage/Tool/CoverageTests.cs]: please don't remove this space, it is present on purpose

File: Source/Codecov.Tests/Coverage/Tool/CoverageTests.cs [hunk 2]
```diff
@@ -10,19 +10,19 @@ namespace Codecov.Tests.Coverage.Tool
...
         public void CoverageReport_Should_Be_Empty_If_The_File_Does_Not_Exits(string fileData)
         {
             // Given
-            var options = Substitute.For<ICoverageOptions>();
+            ICoverageOptions options = Substitute.For<ICoverageOptions>();
```
Comment (AdmiringWorm) [inline Source/Codecov.Tests/Coverage/Tool/CoverageTests.cs]: please don't change the existing style used in the library

File: Source/Codecov/Coverage/Tool/Coverage.cs [hunk 3]
```diff
@@ -21,15 +22,45 @@ public Coverage(ICoverageOptions coverageOptions)
 
         private ICoverageOptions CoverageOptions { get; }
 
-        private static bool VerifyReportFile(string path)
+        private static bool VerifyReportFile(string path, out IEnumerable<string> expandedPath)
```
Comment (AdmiringWorm) [inline Source/Codecov/Coverage/Tool/Coverage.cs]: Not sure if I agree with using an out parameter here.
The method should only verify the files, not actually get the files.

File: Source/Codecov/Coverage/Tool/Coverage.cs [hunk 4]
```diff
@@ -21,15 +22,45 @@ public Coverage(ICoverageOptions coverageOptions)
...
             }
 
-            if (!File.Exists(path))
+            if (path.Contains('*')
+                || path.Contains('?')
+                || path.Contains('!')
+                || path.Contains(','))
+            {
+                // Some basic guards
+                if (!path.EndsWith(".xml", StringComparison.OrdinalIgnoreCase))
```
Comment (AdmiringWorm) [inline Source/Codecov/Coverage/Tool/Coverage.cs]: forcing a `.xml` extension on the paths is unfortunately a no go.
codecov don't just support xml reports, but also json, info and more.

**Review by AdmiringWorm — CHANGES_REQUESTED** [review]

File: Source/Codecov/Coverage/Tool/Coverage.cs (hunk 3 above)
Comment (hanabi1224) [reply Source/Codecov/Coverage/Tool/Coverage.cs]: How about giving it a different name?

File: Source/Codecov.Tests/Coverage/Tool/CoverageTests.cs (hunk 2 above)
Comment (hanabi1224) [reply Source/Codecov.Tests/Coverage/Tool/CoverageTests.cs]: It was new vs2019 code formatter :)

Comment (AdmiringWorm) [reply Source/Codecov.Tests/Coverage/Tool/CoverageTests.cs]: Ah, I see.
Is there any configuration file that can be added to the repository that overrides the these style settings? I'm working in Linux, so I don't use Visual Studio 2019 unfortunately.

File: Source/Codecov/Coverage/Tool/Coverage.cs (hunk 3 above)
Comment (AdmiringWorm) [reply Source/Codecov/Coverage/Tool/Coverage.cs]: I was thinking more of separating the globbing logic into a different method that would handle the expanding of the correct file that will be matched.

But I can look into this at a later time.

**Review by AdmiringWorm — APPROVED** [review]
Thank you for updating. I will probably make some additional changes before the next release of codecov-exe (no ETA of this release, there is some other things I want to include in that release, and time is limited for the next couple of months).

**Comment by AdmiringWorm** [comment]
Your changes have been merged, thank you for your contribution :smiley:

**Pull request merged**